--- ours/common/src/main/java/io/homo/superresolution/api/registry/BackendGroup.java
+++ friend/common/src/main/java/io/homo/superresolution/api/registry/BackendGroup.java
@@ -18,8 +18,6 @@
 
 package io.homo.superresolution.api.registry;
 
-import io.homo.superresolution.api.registry.framegeneration.FrameGenerationDescription;
-import io.homo.superresolution.api.registry.lowlatency.LowLatencyDescription;
 import net.minecraft.network.chat.Component;
 
 import java.util.Objects;
