--- ours/neoforge/src/main/java/io/homo/superresolution/neoforge/mixin/compat/iris/GlFramebufferMixin.java
+++ friend/neoforge/src/main/java/io/homo/superresolution/neoforge/mixin/compat/iris/GlFramebufferMixin.java
@@ -32,11 +32,6 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import static org.lwjgl.opengl.GL46.*;
-
-#if MC_VER >= MC_26_3
-import com.mojang.renderpearl.api.textures.GpuTexture;
-import com.mojang.renderpearl.backend.opengl.GlTexture;
-#endif
 
 @Mixin(value = GlFramebuffer.class, remap = false)
 public abstract class GlFramebufferMixin extends GlResource {
@@ -81,15 +76,7 @@
 
     #else
     @Inject(method = "addDepthAttachment", at = @At("RETURN"))
-    private void checkFboCompleteness(
-            #if MC_VER >= MC_26_3
-            GpuTexture
-            #else
-            com.mojang.blaze3d.textures.GpuTexture
-            #endif
-            texture,
-            CallbackInfo ci
-    ) {
+    private void checkFboCompleteness(com.mojang.blaze3d.textures.GpuTexture texture, CallbackInfo ci) {
         int status = getStatus();
         if (status != GL_FRAMEBUFFER_COMPLETE) {
             SuperResolution.LOGGER.error("Incomplete FBO, code: {}", status);
@@ -97,15 +84,7 @@
     }
 
     @Inject(method = "addDepthAttachment", at = @At("HEAD"))
-    public void addDepthAttachment(
-            #if MC_VER >= MC_26_3
-            GpuTexture
-            #else
-            com.mojang.blaze3d.textures.GpuTexture
-            #endif
-            texture,
-            CallbackInfo ci
-    ) {
+    public void addDepthAttachment(com.mojang.blaze3d.textures.GpuTexture texture, CallbackInfo ci) {
         if (super_resolution$currentDepthAttachmentType != 0) {
             IrisRenderSystem.framebufferTexture2D(
                     getGlId(),
@@ -117,15 +96,7 @@
             );
         }
 
-        super_resolution$currentDepthAttachmentType = super_resolution$detectAttachmentType(
-                TextureInfoCache.INSTANCE.getInfo((
-                        #if MC_VER >= MC_26_3
-                        (GlTexture) texture
-                        #else
-                        (com.mojang.blaze3d.opengl.GlTexture) texture
-                        #endif
-                ).glId()).getInternalFormat()
-        );
+        super_resolution$currentDepthAttachmentType = super_resolution$detectAttachmentType(TextureInfoCache.INSTANCE.getInfo(((com.mojang.blaze3d.opengl.GlTexture) texture).glId()).getInternalFormat());
     }
     #endif
 
