--- ours/native/cpp/SRNativeFSR4/include/sr/fsr4/sr_provider.h
+++ friend/native/cpp/SRNativeFSR4/include/sr/fsr4/sr_provider.h
@@ -3,7 +3,6 @@
 #include "sr/sr_modules.h"
 
 extern "C" {
-    SR_API SRReturnCode srGetFfxFSR4UpscaleProviders(SRUpscaleProvider *outProvider);
-
-    SR_API SRReturnCode srGetFfxFSR4UpscaleProvidersCount(uint32_t *outCount);
-}
+    SR_API SRReturnCode srGetFfxFSR4UpscaleProviders(SRUpscaleProvider * outProvider);
+    SR_API SRReturnCode srGetFfxFSR4UpscaleProvidersCount(uint32_t * outCount);
+}