--- ours/common/src/main/java/io/homo/superresolution/common/SuperResolutionKeyMapping.java
+++ friend/common/src/main/java/io/homo/superresolution/common/SuperResolutionKeyMapping.java
@@ -37,11 +37,7 @@
 
     public static final KeyMapping OPENGUI_KEYMAPPING = new KeyMapping(
             "key.super_resolution.open_config",
-            #if MC_VER >= MC_26_3
-            InputConstants.Type.KEYBOARD,
-            #else
             InputConstants.Type.KEYSYM,
-            #endif
             InputConstants.KEY_F6,
             CATEGORY
     );
