--- ours/common/src/main/java/io/homo/superresolution/srapi/SRTextureResourceDescription.java
+++ friend/common/src/main/java/io/homo/superresolution/srapi/SRTextureResourceDescription.java
@@ -62,7 +62,7 @@
             case R32UI -> SRSurfaceFormat.R32_UINT;
             case DEPTH32 -> SRSurfaceFormat.R32_TYPELESS;
             case DEPTH32F -> SRSurfaceFormat.D32_SFLOAT;
-            case DEPTH16, DEPTH24_STENCIL8, DEPTH24, DEPTH_COMPONENT, DEPTH32F_STENCIL8 -> SRSurfaceFormat.UNKNOWN;
+            case DEPTH24_STENCIL8, DEPTH24, DEPTH_COMPONENT, DEPTH32F_STENCIL8 -> SRSurfaceFormat.UNKNOWN;
             case R16_SNORM -> SRSurfaceFormat.R16_SNORM;
             case R11G11B10F -> SRSurfaceFormat.R11G11B10_FLOAT;
             case RGBA16 -> SRSurfaceFormat.R16G16B16A16_TYPELESS;
