--- ours/common/src/main/java/io/homo/superresolution/common/mixin/lowlatency/v1_21_1/MinecraftReflexMixin.java
+++ friend/common/src/main/java/io/homo/superresolution/common/mixin/lowlatency/v1_21_1/MinecraftReflexMixin.java
@@ -21,7 +21,7 @@
 
 import com.mojang.blaze3d.systems.RenderSystem;
 import io.homo.superresolution.common.SuperResolution;
-import io.homo.superresolution.api.registry.lowlatency.LowLatencyDescription;
+import io.homo.superresolution.api.registry.LowLatencyDescription;
 import io.homo.superresolution.common.lowlatency.LowLatency;
 import net.minecraft.client.Minecraft;
 import org.spongepowered.asm.mixin.Mixin;
