--- ours/common/src/main/java/io/homo/superresolution/common/mixin/gui/GuiRendererDrawAccessor.java
+++ friend/common/src/main/java/io/homo/superresolution/common/mixin/gui/GuiRendererDrawAccessor.java
@@ -25,15 +25,11 @@
 @Mixin(targets = "net.minecraft.client.gui.render.GuiRenderer$Draw")
 public interface GuiRendererDrawAccessor {
     @Accessor("pipeline")
-#if MC_VER >= MC_26_3
-    com.mojang.renderpearl.api.pipeline.RenderPipeline getPipeline();
-#else
     com.mojang.blaze3d.pipeline.RenderPipeline getPipeline();
-#endif
 }
 
 #else
 @Mixin(targets = "net.minecraft.client.Minecraft")
 public interface GuiRendererDrawAccessor {
 }
-#endif
+#endif