--- ours/common/src/main/java/io/homo/superresolution/common/mixin/core/GLXMixin.java
+++ friend/common/src/main/java/io/homo/superresolution/common/mixin/core/GLXMixin.java
@@ -18,7 +18,6 @@
 
 package io.homo.superresolution.common.mixin.core;
 
-#if MC_VER < MC_26_3
 import com.mojang.blaze3d.platform.GLX;
 import io.homo.superresolution.core.graphics.GraphicsCapabilities;
 import org.spongepowered.asm.mixin.Mixin;
@@ -35,11 +34,3 @@
         GraphicsCapabilities.detectSupportedVersions();
     }
 }
-#else
-import net.minecraft.client.Minecraft;
-import org.spongepowered.asm.mixin.Mixin;
-
-@Mixin(Minecraft.class)
-public class GLXMixin {
-}
-#endif
